### PR TITLE
Server spec

### PR DESCRIPTION
**Pull request by karmanyaahm**
Should SSRF recommendations be in the docs rather than the spec? also,I think the spec will be in its comment period for a few days.

### CODE REVIEW

File: server.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#### 413
+
+The push provider MAY return a 413 if the notification payload is too large. Push providers MUST support a minimum payload size of 4000 bytes. Applications SHOULD NOT rely on push providers supporting larger payloads for critical functionality.
+
+#### 429 
+
+The push provider MAY return a 429 if the endpoint is receiving too many requests. Application servers SHOULD honor limits per endpoint, not host. On receipt of a 429, the application server SHOULD slow down or risk losing notifications.
+
+#### Other 
+Any other HTTP responses mean an unknown error occured and the request MAY be tried later. //when's later?
```
Comment (karmanyaahm) [inline server.md:52]: Should the later part just be left out?

**Comment by p1gp1g** [comment]
> Should SSRF recommendations be in the docs rather than the spec? I think the spec will be in its comment period for a few days.

We can do something like this: https://github.com/UnifiedPush/specifications/blob/main/specifications/dbus.md#appendix-implementation-practices

File: server.md (hunk 1 above)
Comment (p1gp1g) [reply server.md:52]: Do you mean "Any other HTTP responses mean an unknown error occurred." ?

BTW : /occured/occurred/ ^^

Comment (karmanyaahm) [reply server.md]: fixed

Comment (karmanyaahm) [reply server.md]: fixed spelling, and rephrased this. Is the new one better?

File: server.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#### 429 
+
+The push provider MAY return a 429 if the endpoint is receiving too many requests. Application servers SHOULD honor limits per endpoint, not host. On receipt of a 429, the application server SHOULD slow down or risk losing notifications.
+
+#### Other 
+Any other HTTP responses mean an unknown error occurred. The push request MAY be tried later at the application server's discretion.
+
+## Appendix: Implementation practices
+
+### SSRF
```
Comment (p1gp1g) [inline server.md:60]: ### Server Side Request Forgery (SSRF)

We strongly recommend implementing basic SSRF protection on your application :
* Restrict to `http://` and `https://` schemes;
* Resolve the domain name to check the IP address;
* Blacklist loopback range (127.0.0.0/8 and ::1);
* Blacklist other private ranges (eg. 192.168.0.0/24, 10.0.0.0/8, fd3d:6c0e:ba95:7468::/64);
* Do not follow redirections.
(You may need to convert IP to a known form - integer for instance - : 127.1234 is a loopback address)

Comment (karmanyaahm) [reply server.md:60]: The endpoint section says http or https, does that need to be reiterated in the appendix? Same with redirections, 3xx is banned should that be reiterated?

I'll add the note about domains and IPs

Comment (p1gp1g) [reply server.md:60]: > 1. As specified in the endpoint section, push providers should only be accessed over HTTP or HTTPS.

To show that the application need to control it :
1. As specified in the endpoint section, push providers must only be accessed over HTTP or HTTPS : schemes need to be restricted to `http://` and `https://`.

File: server.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+## Endpoint
+
+The endpoint MUST be a IRI (RFC 3987) with a scheme of HTTP or HTTPS that MAY use query parameters, a path, or both to identify the push registration. An endpoint's length MUST be less than or equal to 1000 bytes.
+
+## Discovery
+
+On sending a GET request to the full endpoint URL (including query parameters), the push provider MUST return the following JSON:
+```json
+["UnifiedPush Provider"]
```
Comment (p1gp1g) [inline server.md:13]: ["SERV_1.0.0"] ?

**Pull request merged**